homeassistant/components/assist_pipeline/pipeline.py

```diff
@@ -1352,6 +1361,25 @@ async def execute(self) -> None:
                 assert self.stt_metadata is not None
                 assert stt_processed_stream is not None
 
+                if self.stt_wake_word is not None:
+                    # Avoid duplicate detections by checking cooldown
+                    wake_up_key = f"{self.stt_wake_word.id}"
+                    last_wake_up = self.run.hass.data[DATA_LAST_WAKE_UP].get(
+                        wake_up_key
+                    )
+                    if last_wake_up is not None:
+                        sec_since_last_wake_up = time.monotonic() - last_wake_up
+                        if sec_since_last_wake_up < cooldown_seconds:
+                            _LOGGER.debug(
+                                "Speech to text cancelled to avoid duplicate wake up"
+                            )
+                            raise SpeechToTextAborted
```

Comment:
The error should be named after what occurred and not the result. Maybe `DuplicateWakeWordDetected` ?